lib/ansible/modules/windows/win_shortcut.ps1

```diff
@@ -0,0 +1,52 @@
+#!powershell
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+$ErrorActionPreference = "Stop"
+$params = Parse-Args $args
+$TargetFile = Get-AnsibleParam -obj $params -name "src" -failifempty $true
+$ShortcutFile = Get-AnsibleParam -obj $params -name "dest" -failifempty $true
+$result = New-Object psobject @{
+    changed = $FALSE
+}
+try
+{
+ $Scriptsh = New-Object -COM WScript.Shell
+ $targetPath = $Scriptsh.CreateShortcut($ShortcutFile).TargetPath
+ if(!(Test-Path $TargetFile))
```

Comment:
Maybe do the Test-Path before L29 and L30 to avoid spending time creating the com object and target path reference if $TargetFile doesn't exist.